### PR TITLE
Drivers: add drivers to support RoboSense LiDAR

### PR DESCRIPTION
**Pull request by yfcube**

### CODE REVIEW

File: modules/drivers/robosense/BUILD [hunk 1]
```diff
@@ -0,0 +1,63 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
+load("//tools:cpplint.bzl", "cpplint")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_binary(
+    name = "librobosense_driver_component.so",
+    linkopts = ["-shared"],
```
Comment (storypku) [inline modules/drivers/robosense/BUILD:8]: This line should be removed. See #12349 for a reference.
And, what about introducing a  `ROBOSENSE_COPTS` variable to be shared across the whole BUILD file, just like 
'HESAI_COPTS' in the forementioned PR.

**Review by storypku — COMMENTED** [review]
Thanks again for your PR.  It looks OK to me now. @Capri2014 @xiaoxq 
@jinghaomiao @ycool for final review opinion.

File: modules/drivers/robosense/input/input.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+Input::~Input() {
+  close(this->msop_fd_);
+  close(this->difop_fd_);
+}
+
+int Input::setUpSocket(uint16_t port) {
+  int sock_fd = socket(PF_INET, SOCK_DGRAM, 0);
+  if (sock_fd < 0) {
+    std::cerr << "socket: " << std::strerror(errno) << std::endl;
```
Comment (fengqikai1414) [inline modules/drivers/robosense/input/input.cpp:34]: 请统一使用apollo日志接口 AERROR

File: modules/drivers/robosense/driver/utility.h [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ private:
+  mutable std::mutex m_mutex;
+};
+
+struct Thread {
+  Thread() { start = false; }
+  std::shared_ptr<std::thread> m_thread;
+  std::atomic<bool> start;
+};
+class ThreadPool {
```
Comment (fengqikai1414) [inline modules/drivers/robosense/driver/utility.h:68]: 为什么不使用apollo提供的线程池

File: modules/drivers/robosense/dag/rs128.dag [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        class_name : "RobosenseComponent"
+        config {
+          name : "RS128_Driver"
+          config_file_path : "/apollo/modules/drivers/robosense/conf/rs128.pb.txt"
+        }
+    }
+}
+
+module_config {
+    module_library : "/apollo/bazel-bin/modules/drivers/velodyne/compensator/libvelodyne_compensator_component.so"
```
Comment (fengqikai1414) [inline modules/drivers/robosense/dag/rs128.dag:13]: 路径名不应该是velodyne

File: modules/drivers/robosense/dag/rs16.dag [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        class_name : "RobosenseComponent"
+        config {
+          name : "RS16_Driver"
+          config_file_path : "/apollo/modules/drivers/robosense/conf/rs16.pb.txt"
+        }
+    }
+}
+
+module_config {
+    module_library : "/apollo/bazel-bin/modules/drivers/velodyne/compensator/libvelodyne_compensator_component.so"
```
Comment (fengqikai1414) [inline modules/drivers/robosense/dag/rs16.dag:13]: 不应该使用velodyne

File: modules/drivers/robosense/dag/rs32.dag [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        class_name : "RobosenseComponent"
+        config {
+          name : "RS32_Driver"
+          config_file_path : "/apollo/modules/drivers/robosense/conf/rs32.pb.txt"
+        }
+    }
+}
+
+module_config {
+    module_library : "/apollo/bazel-bin/modules/drivers/velodyne/compensator/libvelodyne_compensator_component.so"
```
Comment (fengqikai1414) [inline modules/drivers/robosense/dag/rs32.dag:13]: 不应用使用velodyne

File: modules/drivers/robosense/dag/rsbp.dag [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        class_name : "RobosenseComponent"
+        config {
+          name : "RSbp_Driver"
+          config_file_path : "/apollo/modules/drivers/robosense/conf/rsbp.pb.txt"
+        }
+    }
+}
+
+module_config {
+    module_library : "/apollo/bazel-bin/modules/drivers/velodyne/compensator/libvelodyne_compensator_component.so"
```
Comment (fengqikai1414) [inline modules/drivers/robosense/dag/rsbp.dag:13]: 不应该使用velodyne

**Review by fengqikai1414 — CHANGES_REQUESTED** [review]

File: modules/drivers/robosense/dag/rs128.dag (hunk 4 above)
Comment (CrazyTomX) [reply modules/drivers/robosense/dag/rs128.dag:13]: image
 您好，我们是参考了apollo文档 how_to_add_a_new_lidar_driver中写的可以复用velodyne的运动补偿模块

**Review by fengqikai1414 — APPROVED** [review]

**Pull request merged**